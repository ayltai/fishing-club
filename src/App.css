.mainContent {
    margin-top : 32px;
}
