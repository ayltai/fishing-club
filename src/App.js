// @flow
import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import AppBar from 'material-ui/AppBar';
import AppDrawer from './components/navigation/AppDrawer';
import AppDrawerStore from './components/navigation/AppDrawerStore';
import AppDrawerItem from './components/navigation/AppDrawerItem';
import injectTapEventPlugin from 'react-tap-event-plugin';
import './App.css';

injectTapEventPlugin();

const createAppDrawerStore = () : AppDrawerStore => {
    const store : AppDrawerStore = new AppDrawerStore(true);

    store.items.push(new AppDrawerItem('Home', 'home'));
    store.items.push(new AppDrawerItem('Location', 'map-marker'));
    store.items.push(new AppDrawerItem('Transportation', 'bus'));

    store.selected = 'Home';

    return store;
};

export default class App extends React.Component {
    _appDrawer : AppDrawerStore;

    constructor(props : Object) {
        super(props);

        this._appDrawer = createAppDrawerStore();
    }

    render() : any {
        return (
            <MuiThemeProvider>
                <div>
                    <AppBar
                        title="Fishing Club"
                        onLeftIconButtonTouchTap={() : void => {
                            this._appDrawer.isOpened = !this._appDrawer.isOpened;
                        }} />
                    <AppDrawer store={this._appDrawer} />
                </div>
            </MuiThemeProvider>
        );
    }
}
