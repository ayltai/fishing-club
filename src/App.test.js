'use strict';

import React from 'react';
import ReactDOM from 'react-dom';
import { Router } from 'react-router';
import createBrowserHistory from 'history/createBrowserHistory';
import App from './App';

it('renders correctly', () => {
    const div = document.createElement('div');
    ReactDOM.render(
        <Router history={createBrowserHistory()}>
            <App />
        </Router>,
        div);

    expect(div.innerHTML).toMatchSnapshot();
});
