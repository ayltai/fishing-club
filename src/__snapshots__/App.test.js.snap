// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `

<div data-reactroot>
  <div style="color: rgba(0, 0, 0, 0.87); background-color: rgb(0, 188, 212); box-sizing: border-box; font-family: Roboto, sans-serif; box-shadow: 0 1px 6px rgba(0, 0, 0, 0.12),
         0 1px 4px rgba(0, 0, 0, 0.12); border-radius: 0px; position: relative; z-index: 1100; width: 100%; display: flex; padding-left: 24px; padding-right: 24px;">
    <button tabindex="0"
            type="button"
            style="border: 10px; box-sizing: border-box; display: inline-block; font-family: Roboto, sans-serif; cursor: pointer; text-decoration: none; margin: 8px 8px 0px -16px; padding: 12px; outline: none; font-size: 0; font-weight: inherit; position: relative; z-index: 1; overflow: visible; width: 48px; height: 48px; background: none;"
    >
      <div>
        <svg viewbox="0 0 24 24"
             style="display: inline-block; color: rgb(255, 255, 255); fill: #ffffff; height: 24px; width: 24px;"
        >
          <path d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z">
          </path>
        </svg>
      </div>
    </button>
    <h1 style="white-space: nowrap; overflow: hidden; text-overflow: ellipsis; margin: 0px; padding-top: 0px; letter-spacing: 0; font-size: 24px; font-weight: 400; color: rgb(255, 255, 255); height: 64px; line-height: 64px;">
      Fishing Club
    </h1>
  </div>
  <div>
    <!-- react-empty: 9 -->
    <div class="drawer"
         style="color: rgba(0, 0, 0, 0.87); background-color: rgb(255, 255, 255); box-sizing: border-box; font-family: Roboto, sans-serif; box-shadow: 0 3px 10px rgba(0, 0, 0, 0.16),
         0 3px 10px rgba(0, 0, 0, 0.23); border-radius: 0px; height: 100%; width: 256px; position: fixed; z-index: 1300; left: 0px; top: 0px; overflow: auto;"
    >
      <div role="presentation"
           style="z-index: 1000; width: 84px;"
      >
        <div role="menu"
             style="padding: 8px 0px 8px 0px; display: table-cell; width: 84px;"
        >
          <div>
            <span role="menuitem"
                  tabindex="0"
                  style="border: 10px; box-sizing: border-box; display: block; font-family: Roboto, sans-serif; cursor: pointer; text-decoration: none; margin: 0px; padding: 0px; outline: none; font-size: 16px; font-weight: inherit; position: relative; z-index: 1; color: rgb(255, 64, 129); line-height: 48px; min-height: 48px; white-space: nowrap; background: none;"
            >
              <div>
                <div style="margin-left: 0px; padding: 0px 16px 0px 72px; position: relative;">
                  <!-- react-text: 17 -->
                  Home
                  <!-- /react-text -->
                  <span class="fa fa-fw fa-home"
                        color="#757575"
                        style="color: rgb(117, 117, 117); position: absolute; font-size: 24px; display: block; height: 24px; width: 24px; top: 0px; margin: 12px; left: 4px;"
                  >
                  </span>
                </div>
              </div>
            </span>
          </div>
          <div>
            <span role="menuitem"
                  tabindex="0"
                  style="border: 10px; box-sizing: border-box; display: block; font-family: Roboto, sans-serif; cursor: pointer; text-decoration: none; margin: 0px; padding: 0px; outline: none; font-size: 16px; font-weight: inherit; position: relative; z-index: 1; color: rgba(0, 0, 0, 0.87); line-height: 48px; min-height: 48px; white-space: nowrap; background: none;"
            >
              <div>
                <div style="margin-left: 0px; padding: 0px 16px 0px 72px; position: relative;">
                  <!-- react-text: 23 -->
                  Location
                  <!-- /react-text -->
                  <span class="fa fa-fw fa-map-marker"
                        color="#757575"
                        style="color: rgb(117, 117, 117); position: absolute; font-size: 24px; display: block; height: 24px; width: 24px; top: 0px; margin: 12px; left: 4px;"
                  >
                  </span>
                </div>
              </div>
            </span>
          </div>
          <div>
            <span role="menuitem"
                  tabindex="0"
                  style="border: 10px; box-sizing: border-box; display: block; font-family: Roboto, sans-serif; cursor: pointer; text-decoration: none; margin: 0px; padding: 0px; outline: none; font-size: 16px; font-weight: inherit; position: relative; z-index: 1; color: rgba(0, 0, 0, 0.87); line-height: 48px; min-height: 48px; white-space: nowrap; background: none;"
            >
              <div>
                <div style="margin-left: 0px; padding: 0px 16px 0px 72px; position: relative;">
                  <!-- react-text: 29 -->
                  Transportation
                  <!-- /react-text -->
                  <span class="fa fa-fw fa-bus"
                        color="#757575"
                        style="color: rgb(117, 117, 117); position: absolute; font-size: 24px; display: block; height: 24px; width: 24px; top: 0px; margin: 12px; left: 4px;"
                  >
                  </span>
                </div>
              </div>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>

`;
